viam_cartographer.go

```diff
@@ -458,12 +458,9 @@ func terminateCartoFacade(ctx context.Context, cartoSvc *CartographerService) er
 		return nil
 	}
 
-	var stopErr error
-	if !cartoSvc.cloudStoryEnabled {
-		stopErr := cartoSvc.cartofacade.Stop(ctx, cartoSvc.cartoFacadeTimeout)
-		if stopErr != nil {
-			cartoSvc.logger.Errorw("cartofacade stop failed", "error", stopErr)
-		}
```

Comment:
Moving cloud story enabled logic from `viam_cartographer.go` to `carto_facade.cc` so that proper state changes can be set and checked still